### PR TITLE
DID experiments and review feedback

### PR DESCRIPTION
**Pull request by trgarrett**

### CODE REVIEW

**Comment by trgarrett** [comment]
Implemented most of @rigidity's feedback

File: clsp/utils.clib [hunk 1]
```diff
@@ -5,22 +5,9 @@
...
-    (if (all (size_b32 parent) (size_b32 puzzlehash) (> amount -1))
-      (sha256 parent puzzlehash amount)
-      (x)
-    )
-  )
-
   ;; return the full puzzlehash for a singleton with the innerpuzzle curried in
   (defun-inline calculate_full_puzzle_hash (SINGLETON_STRUCT inner_puzzle_hash)
-    (curry_hashes_inline (f SINGLETON_STRUCT)
+    (curry_hashes (f SINGLETON_STRUCT)
```
Comment (trgarrett) [inline clsp/utils.clib:10]: Docs say it is sometimes more efficient to use inline. After profiling, realized this wasn't one of those times.

File: clsp/fusion_singleton.clsp.hex
Comment (Rigidity) [inline clsp/fusion_singleton.clsp.hex:1]: Nice reduction, should save on a good bit of cost :P

Comment (trgarrett) [reply clsp/fusion_singleton.clsp.hex:1]: With my test case of 1 A and 2 B NFT's, the combined changes here in this PR went from ~487M to ~477M. We'll definitely get to a point where everyone is counting their costs. Appreciate the suggestions!

**Pull request merged**